### PR TITLE
Set up Travis & relative path support

### PR DESCRIPTION
**Pull request by jbmertens**

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -26,3 +26,5 @@ end
 # Windows does not include zoneinfo files, so bundle the tzinfo-data gem
 gem 'tzinfo-data', platforms: [:mingw, :mswin, :x64_mingw, :jruby]
 
+# Html proofer, per: https://jekyllrb.com/docs/continuous-integration/travis-ci/
```
Comment (jfly) [inline Gemfile:29]: Neat!

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+- 2.3.3
+
+before_script:
+ - chmod +x ./script/cibuild # or do this locally and commit
+
+# Assume bundler is being used, therefore
+# the `install` step will run `bundle install` by default.
+script: ./script/cibuild
+
+# branch whitelist, only for GitHub Pages
```
Comment (jfly) [inline .travis.yml]: Hmm... what's this all about? Will this prevent travis from running on PRs?

**Review by jfly — COMMENTED** [review]
I don't claim to understand all of this, but am excited to see if it works!

If I understand correctly, this looks like travis will run jekyll (which is great). What about actually committing the changes to the `gh-pages` branch, though? Is that for later?

File: .travis.yml (hunk 2 above)
Comment (jbmertens) [reply .travis.yml]: I think it just prevents (or, only allows) builds on specific branches. There's more explanation here:
https://jekyllrb.com/docs/continuous-integration/travis-ci/#3-configuring-your-travis-builds

**Comment by jbmertens** [comment]
> What about actually committing the changes to the gh-pages branch, though? Is that for later?

Yes.

File: .travis.yml (hunk 2 above)
Comment (jfly) [reply .travis.yml]: I see. This sounds like it's for repositories which have completely different things on their different branches (for example, if TNoodle decided to add a website, we might start a new branch with just jekyll in it, no tnoodle code present). All of the branches of this cubingusa/org project are still going to be jekyll projects, so I don't think this option makes sense for us. The documentation you linked to even says it's optional:

> You want to ensure the Travis builds for your site are being run only on the branch or branches which contain your site. [...] The branches directive is completely optional. Travis will build from every push to any branch of your repo if leave it out.

**Pull request closed without merging**